### PR TITLE
coverage: Don't print out unit-test-only and uncovered in one report

### PR DESCRIPTION
**Pull request by def-**
Sorry for the many changes required to get coverage into a good state. https://buildkite.com/materialize/coverage/builds/96 is running to test this.

Because we have to add back lines for unit-test-only-covered, and remove lines for uncovered code, the resulting lines might not be mapped together line by line again as they were before. See https://buildkite.com/materialize/coverage/builds/93#018804a2-19fc-4696-b11d-7e86383bd1f3 for an example of how this can go wrong, surrounding the line:

> debug!("sending pubsub message: {:?}", message);

Since the surrounding lines have a mix of being uncovered and unit-test-only-covered the diff algorithm puts those blocks together differently, and we end up with a hard-to-read coverage report. For now I split this into two separate reports, one for uncovered code, the other for unit-test-covered-only code. An alternative would be to generate the output ourselves (inline in github comments or using `buildkite-agent annotate`), `diff` is definitely not an appropriate tool for this once you have more than one kind of information to be shown.

### Motivation

- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by def-** [comment]
https://buildkite.com/materialize/coverage/builds/95 looks ok now, new state is like this: 
Screenshot 2023-05-10 at 16 53 15
Ready for review.

**Review by philip-stoev — APPROVED** [review]
The report has this line

```
!    /// to the public internet or other unauthorized parties.
```

Which is odd -- a comment line is listed as uncovered ?

https://buildkite.com/materialize/coverage/builds/97

**Comment by def-** [comment]
There is a misaligned coverage in much of this file unfortunately. We have this problem in a few complex files, I think it could be https://github.com/rust-lang/rust/issues/110487 or a similar new issue. The coverage report looks the same, so at least it's a problem of rustcov and not the new tooling I wrote:

Screenshot 2023-05-11 at 10 18 02

File: misc/python/materialize/cli/ci_coverage_pr_report.py [hunk 1]
```diff
@@ -120,7 +123,7 @@ def mark_covered_lines(
                     coverage[file][line_nr] = (coverage[file][line_nr] or 0) + hit
 
 
-def get_report(coverage: Coverage) -> str:
+def get_report(coverage: Coverage) -> Tuple[str, str]:
```
Comment (nrainer-materialize) [inline misc/python/materialize/cli/ci_coverage_pr_report.py:126]: The two try blocks in this method look pretty similar (I think the diff is only the two additional parts of the if statement regarding the lines). Any chances of extracting a generic method to resolve this clone?

File: misc/python/materialize/cli/ci_coverage_pr_report.py [hunk 2]
```diff
@@ -214,22 +226,48 @@ def main() -> None:
...
-    report = get_report(coverage)
+    (uncovered_report, unit_test_only_report) = get_report(coverage)
+
+    test_case = junit_xml.TestCase("Uncovered Lines in PR", "Code Coverage")
+    if len(uncovered_report):
+        print("Uncovered Lines in PR")
+        # Buildkite interprets the +++ and --- chars at the start of line, put
+        # in a zero-width space as a workaround.
+        print(
+            uncovered_report.replace("\n+++", "\n\u200B+++").replace(
```
Comment (nrainer-materialize) [inline misc/python/materialize/cli/ci_coverage_pr_report.py:237]: Minor: Shall we extract a constant with a meaningful name for `\u200B` to make it easier to read?

File: misc/python/materialize/cli/ci_coverage_pr_report.py (hunk 1 above)
Comment (def-) [reply misc/python/materialize/cli/ci_coverage_pr_report.py]: Thanks, had more differences in the beginning, didn't realize I made them exactly the same :D

**Review by nrainer-materialize — APPROVED** [review]

**Pull request merged**